packages/customer/src/services/customer-module.ts

```diff
@@ -64,18 +72,191 @@ export default class CustomerModuleService implements ICustomerModuleService {
+  ) {
+    const customers = await this.customerService_.list(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<CustomerTypes.CustomerDTO[]>(
+      customers,
       {
         populate: true,
       }
     )
   }
+
+  @InjectManager("baseRepository_")
+  async listAndCount(
+    filters: CustomerTypes.FilterableCustomerProps = {},
+    config: FindConfig<CustomerTypes.CustomerDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<[CustomerTypes.CustomerDTO[], number]> {
+    const [customers, count] = await this.customerService_.listAndCount(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return [
+      await this.baseRepository_.serialize<CustomerTypes.CustomerDTO[]>(
+        customers,
+        {
+          populate: true,
+        }
+      ),
+      count,
+    ]
+  }
+
+  async createCustomerGroup(
+    dataOrArrayOfData: CustomerTypes.CreateCustomerGroupDTO,
+    sharedContext?: Context
+  ): Promise<CustomerTypes.CustomerGroupDTO>
+
+  async createCustomerGroup(
+    dataOrArrayOfData: CustomerTypes.CreateCustomerGroupDTO[],
+    sharedContext?: Context
+  ): Promise<CustomerTypes.CustomerGroupDTO[]>
+
+  @InjectTransactionManager("baseRepository_")
+  async createCustomerGroup(
+    dataOrArrayOfData:
+      | CustomerTypes.CreateCustomerGroupDTO
+      | CustomerTypes.CreateCustomerGroupDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const data = Array.isArray(dataOrArrayOfData)
+      ? dataOrArrayOfData
+      : [dataOrArrayOfData]
+
+    const groups = await this.customerGroupService_.create(data, sharedContext)
+
+    if (Array.isArray(dataOrArrayOfData)) {
+      return await this.baseRepository_.serialize<
+        CustomerTypes.CustomerGroupDTO[]
+      >(groups, {
+        populate: true,
+      })
+    }
+
+    return await this.baseRepository_.serialize<CustomerTypes.CustomerGroupDTO>(
+      groups[0],
+      { populate: true }
+    )
+  }
+
+  async addCustomerToGroup(
+    groupCustomerPair: { customer_id: string; customer_group_id: string },
+    sharedContext?: Context
+  ): Promise<{ id: string }>
+
+  async addCustomerToGroup(
+    groupCustomerPairs: { customer_id: string; customer_group_id: string }[],
+    sharedContext?: Context
+  ): Promise<{ id: string }[]>
+
+  @InjectTransactionManager("baseRepository_")
+  async addCustomerToGroup(
+    data:
+      | { customer_id: string; customer_group_id: string }
+      | { customer_id: string; customer_group_id: string }[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<{ id: string } | { id: string }[]> {
+    const groupCustomers = await this.customerGroupCustomerService_.create(
+      Array.isArray(data) ? data : [data],
+      sharedContext
+    )
+
+    if (Array.isArray(data)) {
+      return groupCustomers.map((gc) => ({ id: gc.customer_group_id }))
+    }
+
+    return { id: groupCustomers[0].customer_group_id }
```

Comment:
suggestion: return type is a bit confusing here, could be either customer or group. The client already has this information. Might be better to return void similar to the `removeCustomerFromGroup`